src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,59 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        File file = new File(fromFileName);
+
+        if (!file.exists() || !file.canWrite()) {
+            throw new RuntimeException("Such file does not exist: " + fromFileName);
+        }
+
+        int totalSupply = 0;
+        int totalBuy = 0;
+
+        try {
+            List<String> rows = Files.readAllLines(file.toPath());
+            for (String row : rows) {
+                String[] parts = row.split(",");
+                String operation = parts[0];
+                try {
+                    int value = Integer.parseInt(parts[1]);
+
+                    if (operation.equals("supply")) {
+                        totalSupply += value;
+                    } else if (operation.equals("buy")) {
+                        totalBuy += value;
+                    }
+                } catch (NumberFormatException e) {
+                    throw new NumberFormatException();
```

Comment:
Throwing a new `NumberFormatException` without a message or the original exception as a cause can make debugging difficult. Consider including a message or the original exception.